src/modules/universal-discovery/universal.discovery.module.js

```diff
@@ -4,30 +4,49 @@ import PropTypes from 'prop-types';
 import TabNavItemComponent from './components/tab-nav/tab.nav.item.component';
 import FinderPanelComponent from './components/tab-content/finder.panel.component';
 import SearchPanelComponent from './components/tab-content/search.panel.component';
+import CreatePanelComponent from './components/tab-content/create.panel.component';
+import ContentCreatorComponent from './components/content-creator/content.creator.component';
 import SelectedContentComponent from './components/selected-content/selected.content.component';
 import ContentMetaPreviewComponent from './components/content-meta-preview/content.meta.preview.component';
 import {
     loadContentInfo,
     loadContentTypes,
     findLocationsByParentLocationId,
-    findContentBySearchQuery
+    findContentBySearchQuery,
+    checkCreatePermission
 } from './services/universal.discovery.service';
 
 import './css/universal.discovery.module.css';
 
 const TAB_BROWSE = 'browse';
 const TAB_SEARCH = 'search';
+const TAB_CREATE = 'create';
 
 export default class UniversalDiscoveryModule extends Component {
     constructor(props) {
         super(props);
 
+        let selectedLanguage = props.languages[0];
+
+        this.onLanguageSelected = this.onLanguageSelected.bind(this);
+        this.onContentTypeSelected = this.onContentTypeSelected.bind(this);
+        this.handlePublish = this.handlePublish.bind(this);
+
+        if (props.forcedLanguageCOTF) {
```

Comment:
I believe it should be renamed to something like: `cotfForcedLanguage` or maybe it can be done differently. Because, here youre hardcoding a kind-of-dependency to the COTF which might not exist on a specific instalation.
Do you really need this information here, in UDW module?

Reply:
COTF is part of a UDW module and the configuration must be set on the UDW and then pass to the COTF. 
`Because, here youre hardcoding a kind-of-dependency to the COTF which might not exist on a specific instalation.` - that's why it has a default value `''` which will prevent from breaking in customers instalations